app/src/androidTest/java/ch/epfl/polychef/RecipeTest.java

```diff
@@ -0,0 +1,199 @@
+        rb.setEstimatedCookingTime(50);
+        rb.setRecipeDifficulty(Recipe.Difficulty.VERY_HARD);
+
+        ArrayList<String> instruc = new ArrayList<>();
+        instruc.add("Start by the beginning");
+        List<Ingredient> ingre = new ArrayList<>();
+        ingre.add(new Ingredient("Carrots", 300d, Ingredient.Unit.NONE));
+
+        Recipe.Difficulty.values();
+        Recipe.Difficulty.valueOf("VERY_HARD");
+        Recipe recipe = rb.build();
+        recipe.getRating();
+
+        assertEquals(recipe.getName(), "Chicken fried");
+        assertEquals(recipe.getRecipeInstructions(), Collections.unmodifiableList(instruc));
+        assertEquals(recipe.getIngredients().get(0), Collections.unmodifiableList(ingre).get(0));
+        assertEquals(recipe.getPersonNumber(), 4);
+        assertEquals(recipe.getEstimatedPreparationTime(), 45);
+        assertEquals(recipe.getEstimatedCookingTime(), 50);
+        assertEquals(recipe.getRecipeDifficulty(), Recipe.Difficulty.VERY_HARD);
+        assertEquals(recipe.getEstimatedTotalTime(), 95);
+        assertEquals(recipe.getMiniaturePath(), "/src/default_miniature.png");
+        assertEquals(recipe.getPicturesNumbers(), Arrays.asList(R.drawable.frenchtoast));
+
+        rb.addPicturePath(15);
+        rb.setMiniaturePath("/src/miniature.jpeg");
+        rb.setMiniaturePath("/src/miniature.png");
+        Recipe recipe2 = rb.build();
+
+        assertEquals(recipe2.getMiniaturePath(), "/src/miniature.png");
+        assertEquals(recipe2.getPicturesNumbers(), Arrays.asList(15));
+    }
+
+    @Test
+    public void argumentsReturnedAreUnmodifiable() {
+        RecipeBuilder rb = setStandardRecipe();
+        rb.setRecipeDifficulty(Recipe.Difficulty.INTERMEDIATE);
+        Recipe recipe = rb.build();
+
+        List<Ingredient> ingre = recipe.getIngredients();
+        List<String> instr = recipe.getRecipeInstructions();
+
+
+        Assertions.assertThrows(UnsupportedOperationException.class, () -> ingre.add(new Ingredient("Steaks", 1000d, Ingredient.Unit.GRAM)));
+        Assertions.assertThrows(UnsupportedOperationException.class, () -> instr.add("/src/hello.png"));
+        Assertions.assertThrows(UnsupportedOperationException.class, () -> instr.set(0, "/src/evilChanger.png"));
+
+        //TODO: to implement
+        //Assertions.assertThrows(UnsupportedOperationException.class, () -> ingre.get(0).setQuantity(50));
+    }
+
+    @Test
+    public void changingPersonNumberScalesIngredients() {
+        RecipeBuilder rb = setStandardRecipe();
+        rb.setRecipeDifficulty(Recipe.Difficulty.EASY);
+        Recipe recipe = rb.build();
+
+        Assertions.assertThrows(IllegalArgumentException.class, () -> recipe.scalePersonAndIngredientsQuantities(0));
+
+        recipe.scalePersonAndIngredientsQuantities(2);
+        for(Ingredient ingredient: recipe.getIngredients()){
+            if(ingredient.getName().equals("Carrots")){
+                assertTrue(ingredient.getQuantity() == 150d);
```

Comment:
We should use assertThat or assertEquals for better error message instead of assertTrue (same multiple time in this file)